### PR TITLE
RestClient cleanups; Pass min/max height filters to backend

### PR DESCRIPTION
**Pull request by webmaster128**

### CODE REVIEW

File: packages/bcp/src/cosmwasmconnection.ts [hunk 1]
```diff
@@ -314,32 +316,20 @@ export class CosmWasmConnection implements BlockchainConnection {
       );
     }
 
-    const minHeight = minHeightOptional || 0;
-    const maxHeight = maxHeightOptional || Number.MAX_SAFE_INTEGER;
-
-    if (maxHeight < minHeight) return []; // optional optimization
+    const filter: SearchTxFilter = { minHeight: minHeight, maxHeight: maxHeight };
```
Comment (ethanfrey) [inline packages/bcp/src/cosmwasmconnection.ts]: You make these two fields required now?

Comment (ethanfrey) [reply packages/bcp/src/cosmwasmconnection.ts]: Nevermind, in the tests it looks like missing/undefined fields are fine

File: packages/sdk/src/cosmwasmclient.spec.ts [hunk 2]
```diff
@@ -309,6 +309,87 @@ describe("CosmWasmClient", () => {
         }),
       );
     });
+
+    it("can search by ID and filter by minHeight", async () => {
```
Comment (ethanfrey) [inline packages/sdk/src/cosmwasmclient.spec.ts]: nice tests

File: packages/sdk/src/restclient.spec.ts [hunk 3]
```diff
@@ -169,15 +170,61 @@ describe("RestClient", () => {
...
-    it("works", async () => {
+  // The /auth endpoints
+
+  describe("authAccounts", () => {
+    it("works for unused account without pubkey", async () => {
       pendingWithoutWasmd();
       const client = new RestClient(httpUrl);
-      const info = await client.nodeInfo();
-      expect(info.node_info.network).toEqual(defaultNetworkId);
+      const { result } = await client.authAccounts(unusedAccount.address);
```
Comment (ethanfrey) [inline packages/sdk/src/restclient.spec.ts]: This part is just unrelated test cleanup, right?
Seems solid, but I don't see it to do with min/maxHeight (or am I missing something?)

Comment (ethanfrey) [reply packages/sdk/src/restclient.spec.ts]: Oh, this is due to a bad diff algorithm... the code was here before

File: packages/sdk/src/restclient.spec.ts [hunk 4]
```diff
@@ -245,54 +292,190 @@ describe("RestClient", () => {
...
+      expect(parseInt(result.count, 10)).toEqual(1);
+      expect(parseInt(result.limit, 10)).toEqual(200);
+      expect(parseInt(result.page_number, 10)).toEqual(1);
+      expect(parseInt(result.page_total, 10)).toEqual(1);
+      expect(parseInt(result.total_count, 10)).toEqual(1);
+      expect(result.txs.length).toBeGreaterThanOrEqual(1);
+      expect(result.txs[result.txs.length - 1]).toEqual(posted.tx);
+    });
+
+    it("can filter by tx.hash and tx.minheight", async () => {
```
Comment (ethanfrey) [inline packages/sdk/src/restclient.spec.ts]: Nice

File: packages/sdk/src/restclient.spec.ts (hunk 3 above)
Comment (webmaster128) [reply packages/sdk/src/restclient.spec.ts]: Yeah a lot of diff was created by alphabetically sorting the endpoints
